### PR TITLE
Test/local network snapshot

### PR DESCRIPTION
**Pull request by rhyslbw**
# Context
Our current `cardano-services` integration test DB snapshot is not ideal, given it only contains a subset of the data managed by a tedious selection process to satisfy particular test demands. For this reason, we cannot currently make assertions based on what we know from the tests that have run.

# Proposed Solution
Use the local-network state after the wallet e2e test has run as a source for running integration tests. This is done using a bash script to coordinate the network setup/teardown, test execution, and snapshot creation. While it's not possible to make concrete assertions that transactions appear in hard-coded blocks, additional setup queries should be possible to make similar assertions in an dynamic way.

# Important Changes Introduced
- Upgrades Docker Compose to V2
- The `local-network` database is currently just being loaded alongside `testnet` for demonstration of the workflow.

### CODE REVIEW

**Review by mirceahasegan — APPROVED** [review]
LGTM 🚀

**Review by mkazlauskas — APPROVED** [review]
Looks good, although we probably need to update all jest snapshots too?

**Comment by rhyslbw** [comment]
> Looks good, although we probably need to update all jest snapshots too?

Yeah, I'm going to see if we can switch from using the other type of snapshot completely, then will update the test setup and snapshots and assertions.

File: packages/cardano-services/test/ChainHistory/ChainHistoryHttpService.test.ts [hunk 1]
```diff
@@ -464,26 +443,18 @@ describe('ChainHistoryHttpService', () => {
 
       describe('server and snapshot testing', () => {
```
Comment (rhyslbw) [inline packages/cardano-services/test/ChainHistory/ChainHistoryHttpService.test.ts:444]: This describe block is a bit vague. Can you think of a better name?

File: packages/cardano-services/.env.test [hunk 2]
```diff
@@ -20,3 +20,7 @@ POSTGRES_SSL_CA_FILE=./test/ssl-ca-util.pem
 
 OGMIOS_SRV_SERVICE_NAME=ogmios-test-domain
 RABBITMQ_SRV_SERVICE_NAME=rabbitmq-test-domain
+
+# UNIT TESTS - We need to support both while we complete the migration
+UNIT_TESTS_POSTGRES_CONNECTION_STRING=postgresql://postgres:doNoUseThisSecret!@127.0.0.1:5433/testnet
```
Comment (rhyslbw) [inline packages/cardano-services/.env.test:25]:
```suggestion
INTEGRATION_TESTS_POSTGRES_CONNECTION_STRING=postgresql://postgres:doNoUseThisSecret!@127.0.0.1:5433/testnet
```

File: packages/cardano-services/.env.test [hunk 3]
```diff
@@ -20,3 +20,7 @@ POSTGRES_SSL_CA_FILE=./test/ssl-ca-util.pem
...
+LOCALNETWORK_UNIT_TESTS_POSTGRES_CONNECTION_STRING=postgresql://postgres:doNoUseThisSecret!@127.0.0.1:5433/localnetwork
```
Comment (rhyslbw) [inline packages/cardano-services/.env.test:26]:
```suggestion
LOCALNETWORK_INTEGRATION_TESTS_POSTGRES_CONNECTION_STRING=postgresql://postgres:doNoUseThisSecret!@127.0.0.1:5433/localnetwork
```

File: packages/cardano-services/.env.test [hunk 4]
```diff
@@ -20,3 +20,7 @@ POSTGRES_SSL_CA_FILE=./test/ssl-ca-util.pem
 
 OGMIOS_SRV_SERVICE_NAME=ogmios-test-domain
 RABBITMQ_SRV_SERVICE_NAME=rabbitmq-test-domain
+
+# UNIT TESTS - We need to support both while we complete the migration
```
Comment (rhyslbw) [inline packages/cardano-services/.env.test:24]:
```suggestion
# INTEGRATION TESTS - We need to support both while we complete the migration
```

File: packages/e2e/local-network/scripts/make-babbage.sh [hunk 5]
```diff
@@ -107,7 +107,6 @@ cp templates/babbage/byron-configuration.yaml "${ROOT}/configuration.yaml"
 
 $SED -i "${ROOT}/configuration.yaml" \
      -e 's/Protocol: RealPBFT/Protocol: Cardano/' \
-     -e 's/minSeverity: Info/minSeverity: Debug/' \
```
Comment (iccicci) [inline packages/e2e/local-network/scripts/make-babbage.sh:110]: I guess this is the reason why I wasn't able to set `Info` log level by changing the `node-config.json`!

File: packages/e2e/local-network/scripts/update-pools-certificate.sh [hunk 6]
```diff
@@ -129,7 +129,7 @@ for ((i = 0; i < ${#SP_NODES[@]}; ++i)); do
   cardano-cli stake-pool registration-certificate \
       --cold-verification-key-file "$coldVKey" \
       --vrf-verification-key-file "$vrfKey" \
-      --pool-pledge 500000000 \
+      --pool-pledge 0 \
```
Comment (iccicci) [inline packages/e2e/local-network/scripts/update-pools-certificate.sh:132]: Could somebody please point me a good reading to understand the meaning of this change?
Thank you

Comment (rhyslbw) [reply packages/e2e/local-network/scripts/update-pools-certificate.sh:132]: https://hydra.iohk.io/build/17427421/download/1/delegation_design_spec.pdf
https://github.com/input-output-hk/cardano-ledger

**Comment by rhyslbw** [comment]
Looks good @AngelCastilloB. I've just rerun the CI workflow as it was failing with the `ValueNotConserved` error

**Review by mkazlauskas — APPROVED** [review]
Looks good, just note that this change
> healtcheck of dbSync now waits until ithas synced up to three epochs

will make cardano-services unusable for some time when some testnet is recreated

**Review by mkazlauskas — DISMISSED** [review]

**Review by mkazlauskas — APPROVED** [review]

**Review by mirceahasegan — APPROVED** [review]

**Pull request merged**